### PR TITLE
✨ Make pageViewId64 available to amp-ad in 3p frames

### LINKED ISSUE #34190: I2I: Make pageViewId64 variable available for delayed fetch amp-ad
## Summary

Both `pageViewId` and `pageViewId64` are available for fast-fetch ad networks. However only `pageViewId` is available for delayed-fetch ad networks.

I'd like to make `pageViewId64` also available for delayed-fetch ad networks that render inside a 3p frame.

## Motivation

`pageViewId64` has a higher entropy, and it's very useful for ad serving on sites that may have  thousands of page views per user, where `pageViewId` as a high risk of collision. Actually, this collision is already happening to us, and it's the main motivation to implement this.

`pageViewId64 ` already available in fast-fetch ad networks, like DoubleClick or Adsense.

/cc @ampproject/wg-approvers @ampproject/wg-ads-reviewers @ampproject/wg-components

### PR DESCRIPTION
**Pull request by oliverfernandez**
Fixes #34190

### CODE REVIEW

File: extensions/amp-ad/0.1/amp-ad-3p-impl.js
Comment (calebcordry) [inline extensions/amp-ad/0.1/amp-ad-3p-impl.js:398]: @powerivq any concern with the blocking promise here?

**Review by calebcordry — COMMENTED** [review]
LGTM but let's get another set of eyes as well.

**Review by calebcordry — APPROVED** [review]

**Comment by oliverfernandez** [comment]
Thank you @calebcordry

Do you see any concern @powerivq ?

**Review by powerivq — APPROVED** [review]

**Comment by calebcordry** [comment]
Sorry for the delay on this one, we can merge once all the tests pass. For now it looks like you have some lint errors https://app.circleci.com/pipelines/github/ampproject/amphtml/11750/workflows/7dcff3b5-0023-4dc0-8f29-1c32bfb3f762/jobs/186187

LMK if you have trouble fixing.

**Comment by oliverfernandez** [comment]
I fixed the lint problems @calebcordry 

However, now I get some test failing in `test-amp-context.js` and I don't quite understand why they are failing.

I checked that the same tests are failing in current `main` branch

Could you help me with this?

cc/ @ampproject/build-on-duty

**Comment by calebcordry** [comment]
I went ahead and restarted the tests, agree seems unrelated.

**Comment by calebcordry** [comment]
I restarted a few times and it keeps failing, would you mind syncing to latest main again to see if it helps? Sorry for the hassle, let me know if you get stuck. Thanks

**Review by calebcordry — APPROVED** [review]

**Comment by calebcordry** [comment]
@oliverfernandez you are right, fix is out for review here https://github.com/ampproject/amphtml/pull/35093. Once that is in we should be able to merge.

**Comment by calebcordry** [comment]
Finally passed. Thanks for contributing!

**Pull request merged**
